spec/enumerable_spec.rb

```diff
@@ -0,0 +1,113 @@
+# spec/calculator_spec.rb
+require './enumerable_methods.rb'
+
+describe Enumerable do
+  let(:test) { Array.new([nil, 2.5, 'hi', -7, true]) }
+  let(:arr_i) { Array.new(6) { |i| i + 1 } }
+  let(:arr_s) { Array.new(%w[ant ant ant]) }
+  let(:has) { { a: 10, b: 20 } }
+  let(:range) { (2..10) }
+  let(:proc_case) { proc { |x| x * 4 } }
+
+  describe '#my_each' do
+    it 'returns the elements of an array' do
+      expect(test.my_each.to_a).to eql(test.each.to_a)
+    end
+    it 'returns the elements of an array which presents a block' do
+      expect(test.my_each.to_a { |el| el + 4 }).to eql(test.each.to_a { |el| el + 4 })
+    end
+    it 'returns the elements of a range' do
+      expect(range.my_each.to_a).to eql(range.each.to_a)
+    end
+    it 'returns the elements of a hash' do
+      expect(has.my_each.to_a).to eql(has.each.to_a)
+    end
```

Comment:
## Negative testing state 
- Check if specs cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)) Example: there is a spec for correct usage of the method \[e.g. `it "creates object if all required attributes are passed"`\] and also there is spec for the wrong usage of a method \[e.g. `it "does not create the object if required attributes are missing"`\].

**Example**

```ruby
# POSITIVE
it 'returns the elements of a range' do
      expect(range.my_each.to_a).to eql(range.each.to_a)
    end

# NEGATIVE

it 'does not return the elements of a range' do
      expect(range.my_each.to_a).to eql(another_case)
    end
```